src/model.py

```diff
@@ -855,6 +849,36 @@ def init_optimizer(self):
                 verbose=True,
                 eps=1e-10,
             )
+
+        # make sure non-single-task-specific components are added to optimizers
+        if other_components is not None:
+            for component in other_components:
+                if other_components[component] is not {}:
+                    lr = other_components[component].get("lr")
+                    lr = lr if lr is not None else self.learning_rate
+
+                    # IMN architecture
+                    if component == "shared":
+                        tasks = other_components[component].get("tasks")
+                        tasks = tasks if tasks is not None else []
+
+                        for task in tasks:
+                            optimizers[task].add_param_group(
+                                {"params": self.components[component].parameters(), "lr":lr}
+                            )
+                   
+                    # RACL architecture
+                    elif component == "relations":
+                        for layer in self.components["relations"]:
+                            for task in self.subtasks:
+                                if task in layer:
+                                    try:
+                                        optimizers[task].add_param_group(
+                                            {"params": self.components[component].parameters(), "lr":lr}
+                                        )
+                                    except ValueError:
+                                        "already added this param to this optimizer"
```

Comment:
This is actually wrong. 
`self.components[component][layer]` needs to be added here. 

This current implementation adds all three relation attentions to all the tasks.